packages/flutter_tools/lib/src/doctor.dart

```diff
@@ -629,6 +629,13 @@ class FlutterValidator extends DoctorValidator {
     return null;
   }
 
+  bool _filePathContainsDirPath(String directory, String file) {
+    if (_platform.isWindows) {
+      return file.toLowerCase().startsWith(directory.toLowerCase());
+    }
+    return file.startsWith(directory);
+  }
```

Comment:
Some more concerns:

1. The Flutter root is `C:\flutter` but the resolved Flutter path is `C:\flutterbin`
2. The Flutter root is `C:\flutter` but the resolved Flutter path is `C:\hello\..\flutter`
3. The Flutter root is `C:/flutter` but the resolved Flutter path is `C:\flutter`

Reply:
> Some more concerns:
> 
> 1. The Flutter root is `C:\flutter` but the resolved Flutter path is `C:\flutterbin`

these are different paths, right?

> 2. The Flutter root is `C:\flutter` but the resolved Flutter path is `C:\hello\..\flutter`

I'm pretty sure this would be normalized by `.resolveSymbolicLinksSync()` on line 621.

> 3. The Flutter root is `C:/flutter` but the resolved Flutter path is `C:\flutter`

Good point, i will use `package:path` .canonicalize, which seems like it would cover both case and path separators

Reply:
done

Reply:
> > 1. The Flutter root is C:\flutter but the resolved Flutter path is C:\flutterbin
> these are different paths, right?

Yup this should fail

Reply:
No, good point, lemme add a test case for this